testing/it_sidecar/client/sidecar_client.go

```diff
@@ -34,6 +34,20 @@ var setupCMD = flag.String("setup", "", "the path to the it setup command")
 // ready, and then forwards service logs to test output.  On completion, signals to the it_sidecar
 // to teardown the test namespace
 func (s *K8STestSetup) TestMain(m *testing.M) {
+	s.testMain(m, nil)
+}
+
+// TestMainWithHook will execute the provided setup command, wait for configured pods and services to be
+// ready, forwards service logs to test output, and then runs the provided hook for any custom pre-test suite setup.
+// On completion, signals to the it_sidecar to teardown the test namespace.
+func (s *K8STestSetup) TestMainWithHook(m *testing.M, hook Hook) {
```

Comment:
The more robust approach would be to add 2 callbacks. One should fire when the service is ready ( the `FORWARD` command ), and second when all services are ready ( `READY`  command ). This design will allow to handle the situation when one service ready state depends on the test code interactions with another service.

Reply:
The callback configuration is better to be the part of `K8STestSetup` constructor.